cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -561,111 +564,112 @@ gdv_boolean castBIT_utf8(gdv_int64 context, const char* data, gdv_int32 data_len
     if (compare_lower_strings("false", 5, trimmed_data, trimmed_len)) return false;
   }
   // if no 'true', 'false', '0' or '1' value is found, set an error
-  gdv_fn_context_set_error_msg(context, "Invalid value for boolean.");
+  gdv_fn_context_set_error_msg(context_ptr, "Invalid value for boolean.");
```

Comment:
actually this is a comment not related to this PR. But I feel the error message should have more detail. For instance, 'Invalid value for boolean \<value\>' Just simply stating Invalid value for boolean does not give clarity on what was wrong with the value.